modules/aws/auth.go

```diff
@@ -25,6 +27,24 @@ func NewAuthenticatedSession(region string) (*session.Session, error) {
 	return sess, nil
 }
 
+// CreateAwsSessionFromRole returns a new AWS session after assuming the role
+// whose ARN is provided in roleARN.
+func CreateAwsSessionFromRole(region string, roleARN string) (*session.Session, error) {
```

Comment:
The `NewAuthenticatedSession` method above calls `sess.Config.Credentials.Get()` to check that the credentials are valid and returns an error right away if they are not. This ensures the test fails as early as possible. 

Any reason not to implement a similar mechanic in this method? If so, it would be worth naming it in a similar manner (e.g., `NewAuthenticatedSessionFromRole`).